src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,203 @@
+package duke.task;
+
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.io.Storage;
+
+/**
+ * Class representing the list of tasks
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+    private final Storage storage;
+
+    /**
+     * Constructor of task list. Loads tasks from specified savefile.
+     *
+     * @param storage The storage system that manages saving and loading
+     */
+    public TaskList(Storage storage) {
+        this.storage = storage;
+        this.tasks = storage.load();
+    }
+
+    /**
+     * Adds the task into the taskList
+     * Also saves the taskList and queues a message to be printed
+     *
+     * @param task The task to be added
+     * @return The response
+     * @throws DukeException I/O exception from saving the taskList state
+     */
+    public String addTask(Task task) throws DukeException {
+        tasks.add(task);
+        save();
+
+        return "Got it. I've added this task:\n  " + task.toString() + "\n" + getTaskLengthReport();
+    }
+
+    /**
+     * Marks the specified task as done
+     * Also saves the taskList and queues a message to be printed
```

Comment:
I noticed something and it's a pretty small and nitpicky thing but maybe standardise your statements to end with a fullstop?